tools/tx_pull.py

```diff
@@ -0,0 +1,222 @@
+
+* This script is only valid for OCA projects pushed to Transifex with default
+  naming scheme. This is because there's a reversal operation in the name to
+  get the GitHub repo. It can be easily adapted to get pairs of Transifex slugs
+  with the corresponding GitHub branch.
+* The scan is made downloading each translation file, so it's an slow process.
+  Maybe we can improve this using Transifex statistics (see
+  http://docs.transifex.com/api/statistics/) to check if there is no update
+  in the resource, and comparing with the date of the last commit made by
+  this script (but forces also to check for this commit on GitHub). Another
+  option is to add an argument to provide a date, and check if there is an
+  update for the resource translation beyond that date. As this also needs a
+  call, it has to be tested if we improve or not the speed.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Samuel Lefever
+* Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+
+Maintainer
+----------
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
+"""
+import os.path
+import polib
+import argparse
+from config import read_config
+from github3 import login
+from slumber import API, exceptions
+
+# Read arguments
+parser = argparse.ArgumentParser(
+    description=u'Pull OCA Transifex updated translations to GitHub')
+parser.add_argument(
+    '-p', '--projects', dest='projects', nargs='+',
+    default=[], help='List of slugs of Transifex projects to pull')
+args = parser.parse_args()
+# Read config
+config = read_config()
+gh_username = config.get('GitHub', 'username')
+gh_token = config.get('GitHub', 'token')
+tx_username = config.get('Transifex', 'username')
+tx_password = config.get('Transifex', 'password')
+tx_num_retries = config.get('Transifex', 'num_retries')
+# Connect to GitHub
+github = login(username=gh_username, token=gh_token)
+gh_user = github.user(gh_username)
+gh_credentials = {'name': gh_user.name,
+                  'email': gh_user.email}
+# Connect to Transifex
+tx_url = "https://www.transifex.com/api/2/"
+tx_api = API(tx_url, auth=(tx_username, tx_password))
+
+
+def _load_po_dict(po_file):
+    po_dict = {}
+    for po_entry in po_file:
+        if po_entry.msgstr:
+            key = u'\n'.join(x[0] for x in po_entry.occurrences)
+            key += u'\nmsgid "%s"' % po_entry.msgid
+            po_dict[key] = po_entry.msgstr
+    return po_dict
+
+
+def process_project(tx_project):
+    print "Processing project '%s'..." % tx_project['name']
+    tx_slug = tx_project['slug']
+    oca_project = tx_slug[4:tx_slug.rfind('-', 0, len(tx_slug) - 3)]
```

Comment:
FYI I make this change in a personal branch:
https://github.com/vauxoo-dev/maintainer-tools/commit/88d05331b2660e6ca8fd1d566130fb3c26bf1ac7